software/bismark/methylation_extractor/main.nf

```diff
@@ -0,0 +1,48 @@
+// Import generic module functions
+include { initOptions; saveFiles; getSoftwareName } from './functions'
+
+params.options = [:]
+def options    = initOptions(params.options)
+
+process BISMARK_METHYLATION_EXTRACTOR {
+    tag "$meta.id"
+    label 'process_high'
+    publishDir "${params.outdir}",
+        mode: params.publish_dir_mode,
+        saveAs: { filename -> saveFiles(filename:filename, options:params.options, publish_dir:getSoftwareName(task.process), publish_id:meta.id) }
+
+    conda (params.enable_conda ? "bioconda::bismark==0.23.0" : null)
+    if (workflow.containerEngine == 'singularity' && !params.singularity_pull_docker_container) {
+        container "https://depot.galaxyproject.org/singularity/bismark:0.23.0--0"
+    } else {
+        container "quay.io/biocontainers/bismark:0.23.0--0"
+    }
+
+    input:
+    tuple val(meta), path(bam)
+    path index
+
+    output:
+    tuple val(meta), path("*.bedGraph.gz")         , emit: bedGraph
+    tuple val(meta), path("*.txt.gz")              , emit: methylation_calls
+    tuple val(meta), path("*.cov.gz")              , emit: coverage
+    tuple val(meta), path("*_splitting_report.txt"), emit: report
+    tuple val(meta), path("*.M-bias.txt")          , emit: mbias
+    path "*.version.txt"                           , emit: version
+
+    script:
+    def seqtype  = meta.single_end ? '-s' : '-p'
+    def software = getSoftwareName(task.process)
+    """
+    bismark_methylation_extractor \\
+        --bedGraph \\
+        --counts \\
+        --gzip \\
+        --report \\
```

Comment:
Should these be coming in via `options.args` instead?

Reply:
I *think* these are sensible defaults. By default, the tool produces output per strand so the `--bedGraph` is useful to consolidate this into one file. Also the output can get really massive so it is sensible to put `--gzip`
And `--report` is something that `bismark_summary` and ultimately `MultiQC` needs

Reply:
Just to clarify, `--report` is on by default, but doesn't hurt to specify explicitly. 

The same goes for `--counts`: the coverage files does have the counts, the `bedGraph` file doesn't, and both files are produced for each input if `--bedGraph` is given.

Reply:
Cool beans!